ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/handler/TaskHandler.java

```diff
@@ -58,12 +55,7 @@ public abstract class TaskHandler {
   public abstract List<Runnable> getTasks() throws MetaException;
 
   protected Table resolveTable(String dbName, String tableName) throws MetaException {
-    try {
-      return getMSForConf(conf).getTable(getDefaultCatalog(conf), dbName, tableName);
-    } catch (MetaException e) {
-      LOG.error("Unable to find table {}.{}, {}", dbName, tableName, e.getMessage());
-      throw e;
-    }
+    return CompactorUtil.resolveTable(conf, dbName, tableName);
```

Comment:
How about calling CompactorUtil.resolveTable(conf, dbName, tableName) from initiator directly. 
does taskhandler responsible for all caffein cache related
 tasks?

Reply:
I think it's better to keep a utility method for fetching Table objects in the generic class TaskHandler. Provides an easy way to test things out.

The cache is created in MetastoreCompactorThread and it's passed around to all required TaskHandlers so that it can be used.

See this and its usages -https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/MetadataCache.java

Reply:
For now it is fine to leave it there. But this method is only required because CompactorUtil methods are static. We need to get rid of these static utility methods, so we will be able to mock `CompactorUtil.resolveTable` directly and remove this method.